src/js/core/player.js

```diff
@@ -792,17 +792,53 @@ function name(): {country: string, firstName: string, lastName: string} {
     const fnRand = random.uniform(0, playerNames.first[country][playerNames.first[country].length - 1][1]);
     const firstNameRow = playerNames.first[country].find(row => row[1] >= fnRand);
     if (firstNameRow === undefined) {
-        throw new Error(`Undefined firstNameRow (fnRand=${fnRand}`);
+        throw new Error(`Undefined firstNameRow (fnRand=${fnRand})`);
+    }
+    let firstName = firstNameRow[0];
+
+    // Middle Name, randomly
+    const middle = random.uniform(0, 1);
+    if (middle < 0.05) {
+        const mnRand = random.uniform(0, playerNames.first[country][playerNames.first[country].length - 1][1]);
+        const middleNameRow = playerNames.first[country].find(row => row[1] >= mnRand);
+        if (middleNameRow === undefined) {
+            throw new Error(`Undefined middleNameRow (mnRand=${mnRand})`);
+        }
+        let middleName = middleNameRow[0];
+
+        // abbreviation rules:
+        // 50% will initialize their middle name
+        //   e.g. {Xavier Y. Zed}
+        // 50% will use it fully
+        //   e.g. {Xavier Yanick Zed}
+        const abbrev = random.uniform(0, 1);
+        if (abbrev < 0.5 || country === "USA") {
+            // initialize it
+            middleName = `${middleName.charAt(0)}.`;
+        } else {
+            // use it fully
+        }
+        firstName = `${firstName} ${middleName}`;
     }
-    const firstName = firstNameRow[0];
 
     // Last name
     const lnRand = random.uniform(0, playerNames.last[country][playerNames.last[country].length - 1][1]);
     const lastNameRow = playerNames.last[country].find(row => row[1] >= lnRand);
     if (lastNameRow === undefined) {
-        throw new Error(`Undefined lastNameRow (lnRand=${lnRand}`);
+        throw new Error(`Undefined lastNameRow (lnRand=${lnRand})`);
+    }
+    let lastName = lastNameRow[0];
+
+    // Jr. or III, randomly
+    const son = random.uniform(0, 1);
+    const english = ["USA", "Canada", "Australia", "New Zealand", "England", "Ireland", "Jamaica", "Phillippines"];
+    if (english.indexOf(country) > 0 && son < 0.01) {
+        if (son < 0.004) {
+            lastName = `${lastName} III`;
```

Comment:
Go up to XXIV, so it is theoretically possible to get a value higher than that of any Pope https://en.wikipedia.org/wiki/List_of_popes

And just because we're getting absurd, it shouldn't be a geometric progression. Imagine you're John Smith VII... you're really going to break hundreds of years of tradition? So maybe make the factor lower for Jr (maybe 10%), then 20% for II, 30% for III, 40% for IV, 50% for IV, 60% for V or later.